### PR TITLE
Read project settings from anywhere in the git repo

### PR DESCRIPTION
**Pull request by costinsin**

### CODE REVIEW

**Comment by costinsin** [comment]
* **#11** <a href="https://app.graphite.dev/github/pr/costinsin/gi/11?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @costinsin and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: src/project/settings.rs [hunk 1]
```diff
@@ -65,17 +65,31 @@ impl ProjectSettings {
...
+        };
+
+        std::fs::write(
+            Path::new(&repository_root).join(".git/gi_project_config"),
+            content,
+        )
+        .map_err(|_| {
             eyre!("Failed to save project settings")
                 .suggestion("Check if you have write permissions to the .git directory.")
         })?;
```
Comment (RazvanDire) [inline src/project/settings.rs]:
```suggestion
        .context("Failed to save project settings")
        .suggestion("Check if you have write permissions to the .git directory.")?;
```

Can't you write something like this instead here ?

**Review by RazvanDire — APPROVED** [review]
LGTM

File: src/project/settings.rs (hunk 1 above)
Comment (costinsin) [reply src/project/settings.rs]: Done. Thanks!

**Pull request merged**